src/qml/generic/socialimagecacheitem.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2013 Jolla Ltd.
+ * Contact: Antti seppälä <antti.seppala@jollamobile.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef SOCIALIMAGECACHEITEM_H
+#define SOCIALIMAGECACHEITEM_H
+
+#include <QObject>
+#include <QUrl>
+
+class SocialImageDownloader;
+class SocialImageCacheItemPrivate;
+class SocialImageCacheItem : public QObject
```

Comment:
Hrm.  This could be quite heavy if we have a view with a lot of content, due to all of the signal connections and general QObject slowness.  Do we have a lightweight version of this (ie, a populated struct) which can be used in a model, without requiring QObject instantiation / signal connection / etc in our UI views?